// @nomicfoundation/hardhat-viem-assertions/dist/src/internal/viem-assertions.js
import { balancesHaveChanged } from "./assertions/balances-have-changed.js";
import { emitWithArgs } from "./assertions/emit/emit-with-args.js";
import { emit } from "./assertions/emit/emit.js";
import { revertWithCustomErrorWithArgs } from "./assertions/revert/revert-with-custom-error-with-args.js";
import { revertWithCustomError } from "./assertions/revert/revert-with-custom-error.js";
import { revertWith } from "./assertions/revert/revert-with.js";
import { revert } from "./assertions/revert/revert.js";
export class HardhatViemAssertionsImpl {
    #viem;
    constructor(viem) {
        this.#viem = viem;
    }
    async balancesHaveChanged(txHash, changes) {
        return await balancesHaveChanged(this.#viem, txHash, changes);
    }
    async emit(txHash, contract, eventName) {
        return await emit(this.#viem, txHash, contract, eventName);
    }
    async emitWithArgs(txHash, contract, eventName, args) {
        return await emitWithArgs(this.#viem, txHash, contract, eventName, args);
    }
    async revert(contractFn) {
        return await revert(contractFn);
    }
    async revertWith(contractFn, expectedRevertReason) {
        return await revertWith(contractFn, expectedRevertReason);
    }
    async revertWithCustomError(contractFn, contract, customErrorName) {
        return await revertWithCustomError(contractFn, contract, customErrorName);
    }
    async revertWithCustomErrorWithArgs(contractFn, contract, customErrorName, args) {
        return await revertWithCustomErrorWithArgs(contractFn, contract, customErrorName, args);
    }
}
